hostmap.go

```diff
@@ -397,6 +539,18 @@ func (hm *HostMap) Punchy(ctx context.Context, conn *udp.Conn) {
 	}
 }
 
+func (i *HostInfo) AddRelay(idx uint32, relayIp iputil.VpnIp) {
+	i.Lock()
+	i.relays[relayIp] = struct{}{}
+	i.Unlock()
+}
+
+func (i *HostInfo) GetRemote() *udp.Addr {
+	i.RLock()
+	defer i.RUnlock()
+	return i.remote
```

Comment:
:/ This may cause some problems. I think there are code paths where we hold the hostinfo lock and other paths where we don't, both of which read this value. If we call this function to get the remote, we may try to acquire a lock we've already got.

I think I'll roll this one back, and think deep thoughts in some other PR.